mistakedetection.tests/src/ca/mcgill/sel/mistakedetection/tests/MistakeDetectionPerformanceAnalysis.java

```diff
@@ -613,9 +538,68 @@ private static String sumOfPlaceholdersFormula(String placeholder, int startRow,
     return sb.toString();
   }
 
-  /** Returns the student class diagram with the given identifier. */
-  private static ClassDiagram getStudentClassDiagram(String id) {
-    return cdmFromFile(Paths.get(STUDENT_SOLUTION_DIR, STUDENT_SOLUTION_DIR_NAME_PREFIX + id, CDM_PATH));
+  /** Returns the number of elements in the given class diagram according to the formula used in the experiment. */
+  private static int numElements(ClassDiagram cdm) {
+    final var primitiveTypes = List.of(CDByte.class, CDBoolean.class, CDChar.class, CDDouble.class, CDFloat.class,
+        CDInt.class, CDLong.class, CDString.class, CDAny.class, CDArray.class);
+    final var layoutTypes = List.of(Layout.class, ContainerMapImpl.class, ElementMapImpl.class, LayoutElement.class);
+    int[] resultBox = { 1 };
+    cdm.eAllContents().forEachRemaining(e -> {
+      if (Stream.concat(primitiveTypes.stream(), layoutTypes.stream()).noneMatch(t -> t.isInstance(e))) {
+        if (e instanceof AssociationEnd) {
```

Comment:
This function can be made much nicer with `switch`ing on types once the Java version is upgraded!